async-nats/src/lib.rs

```diff
@@ -539,6 +539,12 @@ impl ConnectionHandler {
                     }
                 }
             }
+            // TODO: we should probably update advertised server list here too.
+            ServerOp::Info(info) => {
+                if info.lame_duck_mode {
+                    self.events.send(ServerEvent::LameDuckMode).await.ok();
```

Comment:
Lets not hyper specialize callbacks, maybe we should just dispatch the info as an event?

```suggestion
                    self.events.send(ServerEvent::Info(info)).await.ok();
```

Reply:
It will eat the first ones, but LDM will be sent only after initial CONNECT and PING from Client.